### PR TITLE
Invalidate sessions feature

### PR DESCRIPTION
**Pull request by tagonzalez**
## Summary

Added feature to invalidate all sessions

## Trello Card

https://trello.com/c/hWFklGzB/28-deshabilitar-todas-las-sesiones

### CODE REVIEW

File: app/middlewares/authenticator.js [hunk 1]
```diff
@@ -12,8 +13,18 @@ module.exports = (req, res, next) => {
     if (err) {
       return next(errors.unauthenticatedUserError(err));
     }
-    logger.info(`Authenticated! Decoded value: ${decode.payload}`);
-    req.decodedValue = decode.payload;
-    return next();
+
+    const iatDate = new Date(decode.iat * 1000);
```
Comment (arinaldi118) [inline app/middlewares/authenticator.js]: we use moment js

File: app/middlewares/authenticator.js [hunk 2]
```diff
@@ -12,8 +13,18 @@ module.exports = (req, res, next) => {
...
+    return models.invalidSessions.findOne({ where: { userId: decode.payload.id } }).then(foundSession => {
+      if (foundSession) {
```
Comment (arinaldi118) [inline app/middlewares/authenticator.js]: `if (foundSession && foundSession.createdAt > iatDate)`

File: app/services/users.js [hunk 3]
```diff
@@ -44,9 +44,18 @@ const listAllUsers = (page, limit) =>
     throw errors.databaseError(err);
   });
 
+const invalidateAllSessions = user =>
+  models.invalidSessions
+    .destroy({ where: { userId: user.id } })
+    .then(() => models.invalidSessions.create({ userId: user.id }))
```
Comment (arinaldi118) [inline app/services/users.js]: You can use `upsert`

**Review by arinaldi118 — CHANGES_REQUESTED** [review]

File: app/middlewares/authenticator.js (hunk 1 above)
Comment (tagonzalez) [reply app/middlewares/authenticator.js]: Changed in latest commits!

File: app/middlewares/authenticator.js (hunk 2 above)
Comment (tagonzalez) [reply app/middlewares/authenticator.js]: Changed in latest commits!

File: app/services/users.js (hunk 3 above)
Comment (tagonzalez) [reply app/services/users.js]: Changed in latest commits!

**Review by arinaldi118 — APPROVED** [review]

**Pull request merged**